src/version.rs

```diff
@@ -47,9 +47,9 @@ impl Version {
 
         // Split the version into semver components.
         let mut iter = version.splitn(3, '.');
-        let major = try!(iter.next().ok_or(error::from_str("missing major version")));
-        let minor = try!(iter.next().ok_or(error::from_str("missing minor version")));
-        let patch = try!(iter.next().ok_or(error::from_str("missing patch version")));
+        let major = try!(iter.next().ok_or_else(|| error::from_str("missing major version")));
+        let minor = try!(iter.next().ok_or_else(|| error::from_str("missing minor version")));
+        let patch = try!(iter.next().ok_or_else(|| error::from_str("missing patch version")));
```

Comment:
This change is fine. It's not really _necessary_, because `error_from_str` is cheap, but it doesn't hurt to make that a closure either.